pkg/instancetype/annotations/annotations_test.go

```diff
@@ -0,0 +1,56 @@
+package annotations_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "kubevirt.io/api/core/v1"
+	apiinstancetype "kubevirt.io/api/instancetype"
+
+	"kubevirt.io/kubevirt/pkg/instancetype/annotations"
+	"kubevirt.io/kubevirt/pkg/libvmi"
+)
+
+var _ = Describe("Annotations", func() {
+	var (
+		vm   *v1.VirtualMachine
+		meta *metav1.ObjectMeta
+	)
+
+	const instancetypeName = "instancetype-name"
+
+	BeforeEach(func() {
+		vm = libvmi.NewVirtualMachine(libvmi.New())
+		vm.Spec.Instancetype = &v1.InstancetypeMatcher{Name: instancetypeName}
+
+		meta = &metav1.ObjectMeta{}
+	})
+
+	It("should add instancetype name annotation", func() {
+		vm.Spec.Instancetype.Kind = apiinstancetype.SingularResourceName
+
+		annotations.Set(vm, meta)
+
+		Expect(meta.Annotations[v1.InstancetypeAnnotation]).To(Equal(instancetypeName))
+		Expect(meta.Annotations[v1.ClusterInstancetypeAnnotation]).To(Equal(""))
```

Comment:
supernit - `BeEmpty()`

Reply:
You didn't change these but thinking about this, isn't this actually a bug with the implementation? We shouldn't be setting the annotation for a cluster instance type at all when using the namespaced type?

Reply:
The result annotations struct does not contain the cluster instance type annotation - it contains just the namespaced one. But I updated the checks to be more precise